### PR TITLE
add channel members page

### LINKED ISSUE #1932: People Widget: link to all members view
As a channel follower, I'd like to be able to see who else is a member of the channel. 

As a channel moderator, I'd like to see who is following my channel, especially people who have recently started following. This will give me a way to acknowledge new followers. 

#### Designs and Mockups

There is a UX video at https://drive.google.com/file/d/1-PBt3UO4wpVuRlL5gFZMS9kQgJGjUOHi/view?usp=sharing

The Members tab itself is mocked up at https://zpl.io/agnKAMO

#### Acceptance Criteria:

- [x] link to "All members" opens "Members" tab
- [ ] add "Members" tab to channel nav
- [x] view of all members in the channel (provided by search API?)
- [ ] default sort is alphabetical
- [ ] sort newest to old and oldest to newer
- [ ] unique URL route for when the members tab is open

#### out of scope

- filter by moderators (how hard is this?)
- sort by comments (how hard is this?)

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #1932 

#### What's this PR do?

This adds a new page in the app which shows all members of a channel (powered by the search index) and adds a link to that view from the 'people' widget.

I also fixed a secondary issue with the click targets for the people widget editing.

#### How should this be manually tested?

go to a channel where you're a moderator and add a new people widget. in the list when you're searching for people you should be able to click anywhere to add the person to the list, not just on the 'add' button. compare with the behavior on master to see what it's fixing.

then, once you've created your widget, you should see a link that says 'See all members', which should take you do a new page, at `/c/${channelName}/members`, which shows you all the people in the channel. You might want to add some fake users in the django shell to fill out the page.

It loads 50 at a time using the infinite scroll component, so make sure that you have more than 50 members so you can test that out.

Note that this doesn't yet implement filtering, as that will require (I think anyway) some changes to the index, so I wanted to leave it off for now in the interest of keeping things simple.

#### Screenshots (if appropriate)

seelalllasdfasdffasdfasdfasdf
seelalllasdfasdff
seelalllasdf

### CODE REVIEW

**Comment by Ferdi** [comment]
Great, "see all members" looks like in the mockup (color, right aligned)

**Comment by aliceriot** [comment]
I updated the display for the "show all members" link:

updated_members_link

and also added a configuration option for showing it or not:

updated_members_form

**Comment by noisecapella** [comment]
Looks great overall. I spotted this bug:
 - Add a new people widget
 - add a user. Do not click the 'show all members link' checkbox
 - click 'update widget' then 'done'

The REST API will return a 400 with a message about a missing `show_all_members_link` field

**Comment by aliceriot** [comment]
cool, thanks for spotting that - I added a default value for the checkbox which seems to fix the issue

File: static/js/containers/SearchPage.js [hunk 1]
```diff
@@ -77,6 +77,11 @@ export class SearchPage extends React.Component<Props, State> {
     }
   }
 
+  componentWillUnmount() {
+    const { clearSearch } = this.props
```
Comment (noisecapella) [inline static/js/containers/SearchPage.js]: Can you add a test covering this, if it's not too difficult to do?

File: static/js/containers/widgets/WidgetListContainer.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ export class WidgetListContainer extends React.Component<Props> {
     })
   }
 
-  updateFrom = (data: WidgetDialogData) => {
+  updateForm = (data: WidgetDialogData) => {
```
Comment (noisecapella) [inline static/js/containers/widgets/WidgetListContainer.js]: :man_facepalming:

**Pull request merged**